tests/components/zwave_js/test_init.py

```diff
@@ -738,63 +842,279 @@ async def test_node_removed(hass, multisensor_6_state, client, integration):
+            "node": multisensor_6_state,
+        },
+    )
+    client.driver.receive_event(event)
     await hass.async_block_till_done()
+
     # Device should still be there after the node was removed
+    device = dev_reg.async_get(dev_id)
     assert device
 
-    event = {"node": new_node}
+    # When the node is replaced, a non-ready node added event is emitted
+    event = Event(
+        type="node added",
+        data={
+            "source": "controller",
+            "event": "node added",
+            "node": {
+                "nodeId": multisensor_6.node_id,
+                "index": 0,
+                "status": 4,
+                "ready": False,
+                "isSecure": "unknown",
+                "interviewAttempts": 1,
+                "endpoints": [
+                    {"nodeId": multisensor_6.node_id, "index": 0, "deviceClass": None}
+                ],
+                "values": [],
+                "deviceClass": None,
+                "commandClasses": [],
+                "interviewStage": "None",
+                "statistics": {
+                    "commandsTX": 0,
+                    "commandsRX": 0,
+                    "commandsDroppedRX": 0,
+                    "commandsDroppedTX": 0,
+                    "timeoutResponse": 0,
+                },
+            },
+        },
+    )
+
+    # Device is still not removed
+    client.driver.receive_event(event)
+    await hass.async_block_till_done()
 
-    client.driver.controller.emit("node added", event)
+    device = dev_reg.async_get(dev_id)
+    assert device
+
+    event = Event(
+        type="ready",
+        data={
+            "source": "node",
+            "event": "ready",
+            "nodeId": node_id,
+            "nodeState": hank_binary_switch_state,
+        },
+    )
+    client.driver.receive_event(event)
     await hass.async_block_till_done()
-    device = dev_reg.async_get(device.id)
-    # assert device is new
+
+    # Old device and entities were removed, but the ID is re-used
+    device = dev_reg.async_get(dev_id)
     assert device
+    assert device == dev_reg.async_get_device(identifiers={(DOMAIN, device_id)})
+    assert device == dev_reg.async_get_device(identifiers={(DOMAIN, hank_device_id)})
+    assert not dev_reg.async_get_device(identifiers={(DOMAIN, multisensor_6_device_id)})
     assert device.manufacturer == "HANK Electronics Ltd."
+    assert device.model == "HKZW-SO01"
+
+    assert not hass.states.get(AIR_TEMPERATURE_SENSOR)
+    assert hass.states.get("switch.smart_plug_with_two_usb_ports")
+
+
+async def test_node_model_change(hass, zp3111, zp3111_state, client, integration):
+    """Test that when a node's model is changed due to an updated device config file, the device and entities are not removed."""
+    # This is not 100% realistic test, since the model change would be seen when the integration is loaded, not via a runtime event. The same registration code path is used though, so it practically similar
```

Comment:
```suggestion
    # This is not 100% realistic test,
    # since the model change would be seen when the integration is loaded,
    # not via a runtime event.
    # The same registration code path is used though, so it's practically similar.
```